app/helpers/sufia/sufia_helper_behavior.rb

```diff
@@ -70,42 +72,52 @@ def number_of_deposits(user)
-    # @param separator [String] (', ') The value to join with.
-    def link_to_facet_list(values, solr_field, empty_message = "No value entered", separator = ", ")
+    # @param values [Array{String}] strings to display
+    # @param solr_field [String] name of the solr field to link to, without its suffix (:facetable)
+    # @param empty_message [String] message to display if no values are passed in
+    # @param separator [String] value to join with
+    # @return [ActiveSupport::SafeBuffer] the html_safe link
+    def link_to_facet_list(values, solr_field, empty_message = "No value entered".html_safe, separator = ", ")
       return empty_message if values.blank?
       facet_field = Solrizer.solr_name(solr_field, :facetable)
       safe_join(values.map { |item| link_to_facet(item, facet_field) }, separator)
     end
 
+    # @param name [String] field name
+    # @param value [String] field value
+    # @param label [String] defaults to value
+    # @param facet_hash [Hash] first argument to Blacklight::SearchState.new
+    # @return [ActiveSupport::SafeBuffer] the html_safe link
+    # @see Blacklight::SearchState#initialize
     def link_to_field(name, value, label = nil, facet_hash = {})
       label ||= value
       params = { search_field: 'advanced', name => "\"#{value}\"" }
       state = search_state_with_facets(params, facet_hash)
       link_to(label, main_app.search_catalog_path(state))
     end
 
-    def index_field_link(options = {})
-      field_name = options[:config][:field_name]
+    # A Blacklight helper_method
+    # @param options [Hash] Blacklight sends :document, :field, :config, :value and whatever else was in options
+    # @option options [Hash] :config including {:field_name => "my_name"}
+    # @option options [Array{String}] :value
+    # @return [ActiveSupport::SafeBuffer] the html_safe link
+    def index_field_link(options)
+      raise ArgumentError unless options[:config] && options[:config][:field_name]
+      name = options[:config][:field_name]
       values = options[:value]
-      safe_join(values.map { |item| link_to_index_field(item, field_name) }, ", ")
-    end
-
-    def link_to_index_field(name, value)
-      params = { search_field: 'advanced', name => "\"#{value}\"" }
-      state = search_state.params_for_search(params)
-      link_to(name, main_app.search_catalog_path(state))
+      safe_join(values.map { |item| link_to_field(name, item, item) }, ", ".html_safe)
     end
 
-    # @param [String,Hash] text either the string to escape or a hash containing the
-    #                           string to escape under the :value key.
+    # @param text [String,Hash] string to escape or a hash containing that string under the :value key.
+    # @param showLink [Boolean]
     def iconify_auto_link(text, showLink = true)
       text = presenter(text[:document]).render_values(text[:value], text[:config]) if text.is_a? Hash
```

Comment:
I need some help with this line, since Blacklight broke SemVer and removed `.render_values`.  @jcoyne recommended `.field_value` as an alternative, and while that method exists, it fails in full tests with:
```
  1) BlacklightHelper render_index_field_value metadata index links description_tesim 
     Failure/Error: text = presenter(text[:document]).field_value(text[:value], text[:config]) if text.is_a? Hash
     
     NoMethodError:
       undefined method `each_pair' for ["This links to http://example.com/ What about that?"]:Array
     # /Users/atz/.rvm/gems/ruby-2.2.0/gems/blacklight-6.3.0/lib/blacklight/configuration/null_field.rb:9:in `initialize'
     # /Users/atz/.rvm/gems/ruby-2.2.0/gems/blacklight-6.3.0/app/presenters/blacklight/index_presenter.rb:66:in `new'
     # /Users/atz/.rvm/gems/ruby-2.2.0/gems/blacklight-6.3.0/app/presenters/blacklight/index_presenter.rb:66:in `block in field_config'
     # /Users/atz/.rvm/gems/ruby-2.2.0/gems/blacklight-6.3.0/app/presenters/blacklight/index_presenter.rb:66:in `fetch'
     # /Users/atz/.rvm/gems/ruby-2.2.0/gems/blacklight-6.3.0/app/presenters/blacklight/index_presenter.rb:66:in `field_config'
     # /Users/atz/.rvm/gems/ruby-2.2.0/gems/blacklight-6.3.0/app/presenters/blacklight/index_presenter.rb:44:in `field_value'
     # ./app/helpers/sufia/sufia_helper_behavior.rb:122:in `iconify_auto_link'
```

This String-or-Hashness of the 1st argument is problematic.